pkg/proxyagent/agent/agent.go

```diff
@@ -370,3 +366,21 @@ func removeDupAndSortServices(services []serviceToExpose) []serviceToExpose {
 
 	return newServices
 }
+
+// The serviceDomain must added because coreDNS is not recursive resolver, and go dns client assume it is.
+// See more details: https://coredns.io/manual/setups/#recursive-resolver; https://github.com/golang/go/blob/6f445a9db55f65e55c5be29d3c506ecf3be37915/src/net/dnsclient_unix.go#L666
+func getServiceDomain() string {
+	apiSvc := "kubernetes.default"
+
+	// In normal case, it will return: `kubernetes.default.svc.cluster.local.`
+	cname, err := net.LookupCNAME(apiSvc)
```

Comment:
I thin this is run on hub?

Reply:
Yes, I forgot this part... we can run this function on a spoke and send the value back to hub. But I guess in that way it would be too complex...

Reply:
Is it possible that we can use clusterlable or status to related this value to a managedcluster CR?

Reply:
who on the spoke will check this?

Reply:
It feels like a meta data of a managed cluster for me, so maybe the klusterlet?